src/graph_compute/numa_scheduling.h

```diff
@@ -138,107 +178,184 @@ static void createChunkData(vertex_t * const nodes, const vid_t cntNodes,
-  }
-}
-
-static void execute_round(const int numRounds, vertex_t * const nodes,
-                          const vid_t cntNodes, scheddata_t * const scheddata) {
-
-  volatile int remainingRounds = numRounds;
-  volatile int remainingPhases = NUM_PHASES;
-  volatile int remainingChunks = cntChunks;
-
-  for (vid_t i = 0; i < scheddata->cntChunks; i++) {
-    scheddata->chunkdata[i].nextIndex = i << CHUNK_BITS;
-  }
-
+static inline void populateWorkerParameters(vertex_t * const nodes,
+                                            const vid_t cntNodes,
+                                            scheddata_t * const scheddata) {
   const int NUM_PHASES = 2;
-  for (int phase = 0; phase < NUM_PHASES; phase++) {
-
+  scheddata->numaSchedInit =
+    static_cast<numaSchedInit_t *>(malloc(sizeof(numaSchedInit_t)*NUMA_WORKERS));
+  numaSchedInit_t * numaSchedInit = scheddata->numaSchedInit;
+  //  a set of NUMA_WORKERS work queues for manages chunks
+  vid_t logChunksPerWorker = logBaseTwoRoundUp(scheddata->numChunksPerWorker + 1);
+  //  the data array used internally by the work queues
+  numaInit_t numaInit(NUMA_WORKERS, 1, static_cast<bool>(NUMA_INIT));
+  scheddata->queueData =
+    static_cast<vid_t *>(numaCalloc(numaInit, sizeof(vid_t),
+                                    NUMA_WORKERS << logChunksPerWorker));
+  for (int i = 0; i < NUMA_WORKERS; i++) {
+    numaSchedInit[i].coreID = i;
+    numaSchedInit[i].numPhases = NUM_PHASES;
+    numaSchedInit[i].cntNodes = cntNodes;
+    numaSchedInit[i].nodes = nodes;
+    numaSchedInit[i].scheddata = scheddata;
+    numaSchedInit[i].workQueue =
+      new (std::nothrow) mrmw_queue_t(&scheddata->queueData[i << logChunksPerWorker],
+                                      logChunksPerWorker, static_cast<vid_t>(-1));
   }
-
 }
 
-static void execute_round(const int numRounds, vertex_t * const nodes,
-                          const vid_t cntNodes, scheddata_t * const scheddata) {
-  #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
-  for (int round = 0; round < numRounds; ++round) {
-    WHEN_DEBUG({
-      cout << "Running chunk round" << round << endl;
-    })
+static inline void init_scheduling(vertex_t * const nodes,
+                                   const vid_t cntNodes,
+                                   scheddata_t * const scheddata) {
+  createChunkData(nodes, cntNodes, scheddata);
+  calculateNodeDependenciesChunk(nodes, cntNodes, scheddata);
+  populateWorkerParameters(nodes, cntNodes, scheddata);
+}
 
-    for (vid_t i = 0; i < scheddata->cntChunks; i++) {
-      scheddata->chunkdata[i].nextIndex = i << CHUNK_BITS;
-    }
+inline void * processChunks(void * param) {
+  numaSchedInit_t * config = static_cast<numaSchedInit_t *>(param);
+  scheddata_t * scheddata = config->scheddata;
+  numaSchedInit_t * numaSchedInit = scheddata->numaSchedInit;
+  // Initialize the chunk
+  bindThreadToCore(config->coreID);
+  static const vid_t sentinel = static_cast<vid_t>(-1);
```

Comment:
Can you make it `SENTINEL` so it's 100% obvious that it's a compile-time constant?